### PR TITLE
[DEVHAS-249] [DEVHAS-256] [DEVHAS-258] CDQ should return git link, Component should check for dup port, SEB preserves base container name for overlays resource

### PR DESCRIPTION
**Pull request by maysunfaisal**
Signed-off-by: Maysun J Faisal <maysunaneek@gmail.com>

#### 249
`CDQ` detection should return github links instead of devfile registry links for a detected sample. 

The reason we do this is because if a devfile registry link is returned for instance https://registry.devfile.io/devfiles/python-basic, the outerloop Devfile Kubernetes file deploy.yaml is not available at https://registry.devfile.io/devfiles/deploy.yaml

But instead of returning a devfile registry link, we return the corresponding github raw link https://raw.githubusercontent.com/devfile-samples/devfile-sample-python-basic/main/devfile.yaml, then deploy.yaml is available at https://raw.githubusercontent.com/devfile-samples/devfile-sample-python-basic/main/deploy.yaml and a `Component` create should be fine.

#### 256 
Looped through the Service and Container ports to check if they were present, before appending

#### 258
Pulls in the latest gitops generator changes updated by @yangcao77 at https://github.com/redhat-developer/gitops-generator/pull/39

### Which issue(s)/story(ies) does this PR fixes:

https://issues.redhat.com/browse/DEVHAS-249
https://issues.redhat.com/browse/DEVHAS-256
https://issues.redhat.com/browse/DEVHAS-258

### PR acceptance criteria:

- [x] Unit/Functional tests

  

- [ ] Documentation 

   

- [ ] Client Impact

  

### How to test changes / Special notes to the reviewer:
- Tests should pass
- Updated tests to check for devfileUrl and dockerfileUrl (249)
- Updated tests to check for duplicate ports (256)
- Tests were updated on gitops gen library (258)
- Update storageLimit & storageRequest tests to increase code coverage

### CODE REVIEW

**Review by flacatus — APPROVED** [review]

File: controllers/componentdetectionquery_controller_test.go [hunk 1]
```diff
@@ -768,8 +776,8 @@ var _ = Describe("Component Detection Query controller", func() {
 		})
 	})
 
-	Context("Create Component Detection Query with springboot repo that has no devfile", func() {
-		It("Should match a devfile", func() {
+	Context("Create Component Detection Query with springboot repo that has devfile", func() {
```
[the remark this answers is not in the document]
Comment (maysunfaisal) [reply controllers/componentdetectionquery_controller_test.go:779]: Also updated line to test that it is returning the devfile in the repo https://github.com/redhat-appstudio/application-service/pull/263/files#diff-cc4a798a709f03ca8d3590af6b15ab884937c99b553375c55ff30299959d07d7R236

**Comment by maysunfaisal** [comment]
Also pulls in gitops library changes by @yangcao77 from https://github.com/redhat-developer/gitops-generator/pull/39

File: controllers/componentdetectionquery_controller_test.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (maysunfaisal) [reply controllers/componentdetectionquery_controller_test.go:781]: updated test to include a component without a devfile

**Review by yangcao77 — APPROVED** [review]
looks good to me

**Review by johnmcollier — APPROVED** [review]

**Review by flacatus — APPROVED** [review]

**Pull request merged**